Libraries/LibWeb/Layout/Node.cpp

```diff
@@ -463,6 +465,67 @@ void NodeWithStyle::apply_style(const CSS::ComputedProperties& computed_style)
                 layer.repeat_y = repeat_value->as_background_repeat().repeat_y();
             }
 
+            if (auto blend_mode_value = value_for_layer(blend_modes, layer_index); blend_mode_value && blend_mode_value->is_keyword()) {
+                auto const num_blend_modes = blend_modes.is_value_list() ? blend_modes.as_value_list().size() : 1;
+
+                if (layer_index >= num_blend_modes) {
+                    layer.blend_mode = CSS::BlendMode::Normal;
+                } else {
```

Comment:
Re your 2nd question: These lines aren't necessary. `value_for_layer()` already deals with the background layers and repeating the list if there aren't enough, so you can just use `blend_mode_value` directly.

If that's not working, let me know and I'll try and figure out what's wrong.